packages/shared/getComponentName.js

```diff
@@ -87,7 +91,19 @@ function getComponentName(type: mixed): string | null {
         if (resolvedThenable) {
           return getComponentName(resolvedThenable);
         }
+        break;
       }
+      case REACT_EVENT_COMPONENT_TYPE:
+        return ((type: any): ReactEventComponent).displayName;
+      case REACT_EVENT_TARGET_TYPE:
+        const eventTarget = ((type: any): ReactEventTarget);
+        if (eventTarget.type === REACT_EVENT_TARGET_TOUCH_HIT) {
```

Comment:
How will this work? Hm... since we have a constant for our own, built-in touch target type- what value will users put there?

Reply:
For user created `EventTarget`s, they'd use their own symbol. That obviously won't be picked up by `getComponentName`, so they'd also need to supply a `displayName` field too.

Reply:
Right, right. My question was unclear. I guess I was really asking- why we need to special case `REACT_EVENT_TARGET_TOUCH_HIT` (to treat it differently than other `REACT_EVENT_TARGET_TYPE`s) given that it wouldn't be possible for us to do a similar thing for custom target-types. I haven't thought a ton about the event system, just seemed like a potential hole.

I agree though that for display name purposes, using an optional `displayName` attribute should be fine.

Reply:
`REACT_EVENT_TARGET_TOUCH_HIT` has special properties that mean it gets handled by the ReactDOM internals differently, specifically that in this other PR to do with hit slop https://github.com/facebook/react/pull/15261.

Reply:
Yeah, I saw that. (I searched the source for it before commenting above.) The question I was trying to ask was if we were comfortable that it's the only target type that will require such special handling. Sounds like the idea is to build onto the API if we find other cases in the future?

Reply:
Sorry, I should have asked if you had search the source before hand. I imagine we'd add more internal types as we go and as we need them and thus add more cases in `getComponentName`. They might also need special code paths in ReactDOM too, some may not.

To be honest, the previous change to this PR wasn't necessary, we could just use `displayName` in all cases and not have a special switch in `getComponentName`. I'm not too bothered on that. What do you think we should do?